tests/cpp/CMakeLists.txt

```diff
@@ -3,6 +3,8 @@ message(STATUS "TEST_ROOT is set to: ${TEST_ROOT}")
 
 include_directories(${TEST_ROOT})
 
-cuda_test(test_copy SRCS ${TEST_ROOT}/cell/test_copy.cu)
-cuda_test(test_gemm SRCS ${TEST_ROOT}/kernels/test_gemm.cu)
-cuda_test(test_scatter_nd SRCS ${TEST_ROOT}/kernels/test_scatter_nd.cu)
+if(WITH_TESTING)
```

Comment:
Add a compilation option in the Makefile that allows users to choose whether to enable or disable tests during the build?

Reply:
The default option currently selected is to build with C++ unit tests enabled, since I think the project is small at the moment, enabling unittest help detect compiling errors. Would you prefer to disable building unit tests by default?

Reply:
It is possible to enable it by default, but I believe that modifying the compilation option to `ENABLE_TEST` would be more understandable for users.

Reply:
I set build with cpp test as the default option.